src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/CompletionIntegrationTests.cs

```diff
@@ -477,6 +450,78 @@ public enum MyEnum
             """);
     }
 
+    [IdeFact, WorkItem("https://github.com/dotnet/razor/issues/11385")]
+    public async Task ProvisionalCompletion_DoesntBreakSemanticTokens()
+    {
+        await TestServices.SolutionExplorer.AddFileAsync(
+            RazorProjectConstants.BlazorProjectName,
+            "Test.razor",
+            """
+            @page "/counter"
+
+            <PageTitle>Counter</PageTitle>
+
+            <h1>Counter</h1>
+
+            <p role="status">Current count: @currentCount</p>
+
+            @DateTime
+
+            <button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+            @code {
+                private int currentCount = 0;
+
+                public void IncrementCount()
+                {
+                    currentCount++;
+                }
+            }
+            """,
+            open: true,
+            ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.WaitForComponentClassificationAsync(ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.PlaceCaretAsync("@DateTime", charsOffset: 1, ControlledHangMitigatingCancellationToken);
+
+        await Task.Delay(500, HangMitigatingCancellationToken);
+
+        TestServices.Input.Send(".");
+        TestServices.Input.Send("n");
```

Comment:
Full disclosure: This test on my machine doesn't type fast enough to repro the issue reliably, but it did do it sometimes.

Reply:
Not sure I follow. The disco colors detection is now in every test essentially, so this test really just tests provisional completion. If any test causes disco colors, that test will fail so the scenario that caused it will be in the name of the test.

Reply:
I'm re-reading my comment, and I think I should expand: This test had flakiness in terms of false-positives (ie, it would sometimes pass) _before my code change_. After my code change I don't expect it to fail or be flaky.